beacon-chain/sync/service.go

```diff
@@ -43,10 +44,16 @@ const seenAttSize = 10000
 const seenExitSize = 100
 const seenProposerSlashingSize = 100
 const badBlockSize = 1000
-
 const syncMetricsInterval = 10 * time.Second
 
-var pendingBlockExpTime = time.Duration(params.BeaconConfig().SlotsPerEpoch.Mul(params.BeaconConfig().SecondsPerSlot)) * time.Second // Seconds in one epoch.
+var (
+	// Seconds in one epoch.
+	pendingBlockExpTime = time.Duration(params.BeaconConfig().SlotsPerEpoch.Mul(params.BeaconConfig().SecondsPerSlot)) * time.Second
+	// seconds to allow processing early blocks.
+	earlyBlockProcessingTolerance = slotutil.MultiplySlotBy(2)
+	// seconds to allow processing early attestations.
+	earlyAttestationProcessingTolerance = slotutil.MultiplySlotBy(2)
```

Comment:
```suggestion
	earlyAttestationProcessingTolerance = params.BeaconNetworkConfig().MaximumGossipClockDisparity
```
We can leave this as the default for now, and discuss the pros/cons on changing attestations here to be 2 slots in a follow up issue/PR. Otherwise it allows a node to process an attestation 2 slots into the future, which deviates from the spec and other clients.